### ISSUE TITLE
Heroku One Click Button Considerations

### ISSUE THREAD

**Issue by n1ru4l** [dungeon-revealer/dungeon-revealer#405]
We currently provide a one-click deploy button for Heroku.

However, in the current state, the application is highly dependent on the file-system. Heroku does use a [ephemeral-filesystem](https://devcenter.heroku.com/articles/dynos#ephemeral-filesystem), which means it is not safe to store data on it.

The proposed solution is using S3 and PostgreSQL instead of SQLite and the FS.

I don't plan adding support for S3 or any other SQL database anytime soon, as it adds additional complexity to the project.

I am not sure whether we should continue advertising the One-Click Button.

**Comment by renzodgc** [comment]
Thanks for the quick assistance! :)

Regarding your question in #402 :
I've used Heroku with PostgreSQL with no issue (with Ruby on Rails). I think the one-click deploy button is really helpful for people who want to use this tool but don't know how to deploy on AWS.

And yeah the ephemeral filesystem is not great, it happened to me multiple times that I had to reupload a map file 😅. But I think the one-click button really adds accessibility!

**Comment by n1ru4l** [comment]
Thanks for the feedback. Maybe there is another service available that is similar to Heroku but in addition, allows specifying or mounting a persistent network volume?

I also had some ideas about providing dungeon-revealer as a service (for the people that do not want to self-host) a while ago. However, I did nothing further yet.

We should definitely add a notice about Heroku not persisting the data.

**Comment by n1ru4l** [comment]
Digital Ocean seems to have something similar which is called One-Click Application Deployments 🤔

**Comment by AlchemicalAgent** [comment]
I agree, it looks like Digital Ocean may be a good solution for a permanent one-click deploy offering. The only drawback I see at the moment is lack of a free hosting tier like Heroku. Their cheapest plan looks like $5/month. That's not a lot considering what it can do but it's still a huge barrier to entry vs. a free solution, even if the free Heroku plan is pretty slow on performance.

**Comment by tirrip** [comment]
Please consider Digital Ocean as one click deployment, or even a guide on how deploy via docker on Digital Ocean and then make it accessible to players and DM. You could put a warning that it's not free. 

On an aside, I have managed to install the one-click docker on Digital Ocean and then run dungeon-revealer on it, but have no idea on how to make it accessible. A simple guide for everyone, or a one-click deployment would broaden the app's possible deploying options?

**Comment by n1ru4l** [comment]
@tirrip Your digital ocean instance should have some public IP attached. Visiting IP:3000 should be sufficient I guess. If you want to use https you need to configure a domain and a certificate https://dnsmap.io/articles/how-to-set-up-a-hostname-with-digitalocean

**Comment by swingtempo** [comment]
Yes, please do not remove the one-click support. I think it makes the tool more accessible, for example I (who have never used Heroku before) would never have tried out Dungeon Revealer without it. I think updating the text of the one click deployment button emphasizing that the data is ephemeral would help.
